src/main/kotlin/nmcp/NmcpAggregationExtension.kt

```diff
@@ -0,0 +1,71 @@
+package nmcp
+
+import gratatouille.GExtension
+import nmcp.internal.configureAttributes
+import nmcp.internal.nmcpConsumerConfigurationName
+import nmcp.internal.task.registerPublishTask
+import org.gradle.api.Action
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Zip
+
+@GExtension(pluginId = "com.gradleup.nmcp.aggregation")
+open class NmcpAggregationExtension(private val project: Project) {
+    internal val spec = project.objects.newInstance(NmcpSpec::class.java)
+
+    internal val consumerConfiguration = project.configurations.create(nmcpConsumerConfigurationName) {
+        it.isCanBeResolved = true
+        it.isCanBeConsumed = false
+
+        it.configureAttributes(project)
+    }
+
+    val zipTaskProvider = project.tasks.register("zipAggregationPublication", Zip::class.java) {
+        it.archiveFileName.set("publicationAggregated.zip")
```

Comment:
The names are do not seem coherent (yet I'm far from being a native speaker).

WDYT of going with `aggregatedPublication.zip` and `zipAggregatedPublication` task name?

Reply:
Great callout! Even `aggregatedPublication` is a bit weird since there is no proper `Publication` for the aggregation.

[I just went with `aggregation` everywhere](https://github.com/GradleUp/nmcp/pull/29/commits/3b41d9e39860fe54af5f5379bfbd10f71078a5f1), let me know what you think.